### ISSUE TITLE
Trakt sync successful but whatched episodes don't show up on trakt

### ISSUE THREAD

**Issue by valz77** [razzeee/script.trakt#531]
## Expected Behavior
When an episode is scrobbled, or when i run sync manually from kodi, the episodes appear as watched on trakt

## Current Behavior
Since updating to Kodi 19 trakt seems to work correctly, but info about watched episodes doesn't show up on trakt. Every time i run sync manually, it detects the same amount of episodes to sync, it says it has updated them correctly, but nothing happens. According to the log file everything is posted correctly to trakt. I tried resetting the add-on and logging in again but nothing changes

## Your Environment

* Version used: 3.3.5
* Kodi version used: 19.0
* Operating System and version: Android TV 8

[kodi_trakt.log](https://github.com/trakt/script.trakt/files/6063572/kodi_trakt.log)

**Comment by razzeee** [comment]
Yeah, it detecting the same episodes to sync basically means, that they didn't get written, I guess. You should probably contact trakt support first. Log looks good from my POV.

**Comment by patel-jeel92** [comment]
I second this. I am running into the same issue as well on Kodi 19 and Trakt v 3.3.5

**Comment by rijsab** [comment]
Yes same issue with Pi4 Kodi 19.1 LibreElec. Not working and no purpose to use Trakt if this is not syncing to actual Trakt account.

**Comment by razzeee** [comment]
As stated contact trakt or post your logs...

**Comment by rijsab** [comment]
It is not a Trakt issue.. I have tested with Plex and everything is working. Seems only not working for Trakt in Kodi Matrix..

**Comment by razzeee** [comment]
Let's get a logfile then https://github.com/trakt/script.trakt#creating-logfiles
(running matrix daily with trakt working fine...)

**Comment by rijsab** [comment]
Here you go pal,

http://ix.io/3rCV

I have tried on both Mateix and Leia now and same.. Trakt website is showing what I am currently playing but will not show up in my history or progress.

**Comment by razzeee** [comment]
That log doesn't show you running anything trakt related, did you check out the link above?

**Comment by rijsab** [comment]
Ok, the log seems to big to upload. 

I managed to get it work but only when completing watching an episode to the end. If I stop it at the middle it wont show up in Trakt progress.. is this known behaviour? Is there a minimum run time for an episode before showing up on Trakt?

**Comment by rijsab** [comment]
I have tried several episodes and they only sync to Trakt history and progress when watching the entire episode. I have forwarded the play time to the end and everything seems to work. So only if the episode is stopped before the end it does not show up in Trakt even when the scrobble notification shows up and Trakt website is showing the episode is playing.

**Comment by razzeee** [comment]
yes, there's a threshold at about 80% I think. can't remember if we rely on kodi's logic or a trakt logic.

**Comment by rijsab** [comment]
Is this something which can be lowered to lets say 10% otherwise we only keep track of fully watched episodes instead of the ones which are still in progress? Would be really helpful.. I can change for personal use if you can direct me to the correct files?

**Comment by razzeee** [comment]
I think your mixing up stuff/features. It sounds a lot, like you want the "in progress" feature (check your settings) and not watched. Watched is only for stuff you've finished, in progress keep track of stuff well, not finished. I don't think trakt has an interface for that on their website.

**Comment by rijsab** [comment]
I have the sync progress of episodes to kodi on On.. basically I want to see which episodes are fully watched and which episodes are In Progress (lets say only 25% of an episode has been watched) and I want to resume this episode again at a later stage.

**Comment by razzeee** [comment]
should be done by kodi out of the box. trakt is only useful in this case, if you loose your data or are running multiple machines and want to sync them.

use watched/in progress in the default kodi skin.

**Comment by rijsab** [comment]
It used always to work as described with full watched and in progress. I know Kodi is doing this but using Trakt to sync to other devices as well.

**Comment by gnoling** [comment]
I stopped using this addon a while back but was coming back to see if I could find an issue where the problem I was having was present/maybe fixed (sorry, I really should've pursued it through an issue report myself). This sounds similar so I'll add my findings from back then before I had moved on.

While I don't have a log now, at the time I attempted to debug the issue for a little while. What I was seeing was that while some of the addon supported re-requesting due to rate-limiting on trakt's end, the watched status syncing wasn't handling it properly. 

What happens is the watched sync operates in batches, pushing a batch at a time until everything is synced. Early batches go through fine, then later ones fail due to the rate-limiting -- I believe it should auto-retry when this happens, but it actually fails here. On the trakt side, the symptoms present as _some_ watched statuses updating early on near the beginning of the video library, then no later ones. In future syncs, it just looks like nothing is updating because the successful batches were for the episodes that succeeded in the past, and therefor just looks like nothing changes.

**Comment by cboRD181** [comment]
> I stopped using this addon a while back but was coming back to see if I could find an issue where the problem I was having was present/maybe fixed (sorry, I really should've pursued it through an issue report myself). This sounds similar so I'll add my findings from back then before I had moved on.
> 
> While I don't have a log now, at the time I attempted to debug the issue for a little while. What I was seeing was that while some of the addon supported re-requesting due to rate-limiting on trakt's end, the watched status syncing wasn't handling it properly.
> 
> What happens is the watched sync operates in batches, pushing a batch at a time until everything is synced. Early batches go through fine, then later ones fail due to the rate-limiting -- I believe it should auto-retry when this happens, but it actually fails here. On the trakt side, the symptoms present as _some_ watched statuses updating early on near the beginning of the video library, then no later ones. In future syncs, it just looks like nothing is updating because the successful batches were for the episodes that succeeded in the past, and therefor just looks like nothing changes.

@razzeee hey, just wondering if this has been looked into? It seems to match what I'm experiencing.

**Comment by razzeee** [comment]
I don't think so, but then again, the statement sounds kind of weird, at least if I don't misremember the logic. I wouldn't expect it to try to upload the same episodes on subsequent tries, we diff locally and only send what's needed.
